### DISCUSSION TITLE
Implement DFlat like TUI windowing in FTXUI

### DISCUSSION THREAD

**Discussion by russkel** [ArthurSonzogni/FTXUI#921] [Q&A]
Hi there,

I am wondering if this is a good/bad idea: implementing DFlat (https://github.com/ghaerr/dflat) styled windowing in FTXUI?

I imagine some of this could be done by splitting hard-coded styling out of the widgets?

![image](https://github.com/user-attachments/assets/fda6e28e-cd7a-4160-921b-ab2ae10affd6)

**Comment by ArthurSonzogni** [comment]
Yes, you should be able to customize the `Window` using:
https://github.com/ArthurSonzogni/FTXUI/blob/fbd56cdf4356e00dcb607314e62800e90722906a/include/ftxui/component/component_options.hpp#L262

Maybe the scrollbars would be more difficult.

**Reply by russkel** [reply]
Thanks for that. I might give it a shot. I saw in the textbox widgets that it creates scrollbars when required. Does this "scroll" support extend to layouts as well?
